spec-tests/src/test/java/org/candlepin/spec/content/OwnerContentSpecTest.java

```diff
@@ -1157,4 +958,1038 @@ public void shouldPermitSafeParallelModification() throws Exception {
+                .toList();
+        }
+
+        private Predicate<ProductInfo> buildOwnerProductPredicate(OwnerDTO owner) {
+            return pinfo -> pinfo.owner() == null || owner.getKey().equals(pinfo.owner().getKey());
+        }
+
+        private Predicate<ProductInfo> buildContentLabelPredicate(List<String> labels) {
+            return pinfo -> pinfo.content()
+                .filter(content -> labels.contains(content.getLabel()))
+                .findAny()
+                .isPresent();
+        }
+
+        @Test
+        public void shouldAllowQueryingWithoutAnyFilters() {
+            OwnerDTO owner = this.owners.get(1);
+
+            List<String> expectedCids = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .flatMap(ProductInfo::content)
+                .map(ContentDTO::getId)
+                .toList();
+
+            List<ContentDTO> output = this.adminClient.ownerContent()
+                .getContentsByOwner(owner.getKey(), null, null, INCLUSION_INCLUDE, INCLUSION_INCLUDE);
+
+            assertThat(output)
+                .isNotNull()
+                .map(ContentDTO::getId)
+                .containsAll(expectedCids)
+                .doesNotContainAnyElementsOf(this.getUnexpectedIds(expectedCids));
+        }
+
+        @Test
+        public void shouldDefaultToActiveExclusiveCustomInclusive() {
+            OwnerDTO owner = this.owners.get(1);
+
+            List<String> expectedCids = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .filter(ProductInfo::active)
+                .flatMap(ProductInfo::content)
+                .map(ContentDTO::getId)
+                .toList();
+
+            List<ContentDTO> output = this.adminClient.ownerContent()
+                .getContentsByOwner(owner.getKey(), null, null, null, null);
+
+            assertThat(output)
+                .isNotNull()
+                .map(ContentDTO::getId)
+                .containsAll(expectedCids)
+                .doesNotContainAnyElementsOf(this.getUnexpectedIds(expectedCids));
+        }
+
+        @Test
+        public void shouldFailWithInvalidOwners() {
+            assertNotFound(() -> this.adminClient.ownerContent()
+                .getContentsByOwner("invalid_owner", null, null, null, null));
+        }
+
+        @Test
+        public void shouldAllowFilteringOnIDs() {
+            // Randomly seeded for consistency; seed itself chosen randomly
+            Random rand = new Random(58258);
+
+            OwnerDTO owner = this.owners.get(1);
+
+            List<String> pids = this.productMap.values()
+                .stream()
+                .filter(elem -> rand.nextBoolean())
+                .map(ProductInfo::id)
+                .toList();
+
+            List<String> expectedCids = this.productMap.values()
```

Comment:
This test is using the expected output as the input list, which lowers the value of the test. The first list (pids) is supposed to be a collection of input content IDs we're filtering on. The second is the expected output which is filtered down by taking that input list and removing anything that isn't from the org.

This should be updated such that we're pulling content IDs for the first list and then building our expected list from that.

Additionally, you should probably respin/check the seed to ensure that it's picking a reasonable chunk of data from the set. If it's biased toward removing too much or too little, the test trends toward edge cases.

Reply:
Updated the test and I also updated the seed. I was able to find a seed to get more distributed data:

[g-content-2a, g-content-2b, g-content-3a, g-content-3b, g-content-1a, g-content-1b, o2-content-1b, o2-content-1a]